packages/eslint-plugin/tests/rules/no-unsafe-enum-comparison.test.ts

```diff
@@ -221,16 +221,6 @@ ruleTester.run('no-unsafe-enum-comparison', rule, {
       declare const right: string | Vegetable2;
       left === right;
     `,
-    `
-      enum Vegetable {
-        Asparagus = 'asparagus',
-        Beet = 'beet',
-        Celery = 'celery',
-      }
-      type WeirdString = string & { __someBrand: void };
-      declare const weirdString: WeirdString;
-      Vegetable.Asparagus === weirdString;
-    `,
```

Comment:
[Testing] Could you keep this case in the file, please? IMO it's good to also check the case of a property like `__someBrand`. Sometimes empty object types in an intersection are handled differently on the TypeScript side.

Reply:
I don't understand the test. Specifically, consider the following code:

```
type String = string;
declare const string: String;
Vegetable.Asparagus === string;
```

Currently, this code flags the "no-unsafe-enum-comparison" rule, which seems correct, because it is not possible for the `String` type to contain an enum. So far so good.

So why would this logic change we expand the scope to intersection? Even though `WeirdString` is an intersection, the same logic should apply. In other words, it is not possible for the intersection of `WeirdString` to *ever* contain a vegetable, so therefore we should want the rule to flag this.

Reply:
It's a [Branded Type](https://github.com/LearningTypeScript/site/pull/135). Folks do this sometimes and use strings as the actual runtime value.

Reply:
Right, I used branded types all the time myself. But how does testing for a branded type make sense with respect to what I wrote in my previous post?

Reply:
I just want to make sure that the logic _doesn't_ change.

Reply:
I don't know what you mean by "logic". After my PR, the rule will trigger in more situations than it did before, which causes the test case failure.

Reply:
By "logic" I mean the rule's handling of this case. As in, I want to make sure that branded types are still something in the tests. 

👍 you do have them separately though so I think this thread is resolved. Thanks 😄